### DISCUSSION TITLE
select with 20k data

### DISCUSSION THREAD

**Discussion by kumard3** [ariakit/ariakit#3672] [Questions] [answered]
i am creating an application in with i have to put a state, cities select , and it is going more than 20k  ,and i have tried virtualized select , but the ui is not that good and it is having some issue , so is it possible to use the select or can i add my custom dropdown ?

**Comment by diegohaz** [comment, marked as answer]
> but the ui is not that good and it is having some issue

What's the issue?

**Reply by kumard3** [reply]
this is the main issue i am having with the current package i am using .
https://github.com/ariakit/ariakit/assets/62591080/2d85e43a-7680-4659-91fd-f55daa82da70

**Reply by diegohaz** [reply]
Could you please open a new issue with a repro in StackBlitz or CodeSandbox?

**Reply by diegohaz** [reply]
Here's an example using the experimental `SelectRenderer` component to render 20k items: https://stackblitz.com/edit/sgftvk?file=select%2Findex.tsx&theme=dark

**Reply by kumard3** [reply]
Thanks for the reply, this example is really helpful, I am wording of it is possible to to pass a custom drop-down component to the select component ? 
By the way, the example is really helpful and I am definitely going to use it .
